lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,363 @@
+    for (var subscription in _subscriptions) {
+      subscription.cancel();
+    }
+    _subscriptions.clear();
+  }
+
+  /// A callback called when the engine begins running [liveTest].
+  void _onTestStarted(LiveTest liveTest) {
+    if (liveTest.suite is! LoadSuite) {
+      if (!_stopwatch.isRunning) _stopwatch.start();
+
+      // The engine surfaces load tests when there are no other tests running,
+      // but because the expanded reporter's output is always visible, we don't
+      // emit information about them unless they fail.
+      _subscriptions.add(liveTest.onStateChange
+          .listen((state) => _onStateChange(liveTest, state)));
+    } else if (_engine.active.length == 1 &&
+        _engine.active.first == liveTest &&
+        liveTest.test.name.startsWith("compiling ")) {}
+
+    _subscriptions.add(liveTest.onError
+        .listen((error) => _onError(liveTest, error.error, error.stackTrace)));
+
+    _subscriptions.add(liveTest.onPrint.listen((line) {}));
+  }
+
+  /// Return an ordered list of groups that [liveTest] belongs to
+  List _orderedGroupList(List<Group> listOfGroups) {
+    List separatedList = [];
+    for (int i = listOfGroups.length - 1; i >= 1; i--) {
+      if (i > 1) {
+        separatedList.add(listOfGroups[i]
+            .name
+            .replaceAll(listOfGroups[i - 1].name, '')
+            .trim());
+      } else {
+        separatedList.add((listOfGroups[1].name));
+      }
+    }
+    return (separatedList.reversed).toList();
+  }
+
+  /// A callback called when [liveTest]'s state becomes [state].
+  void _onStateChange(LiveTest liveTest, State state) {
+    if (state.status != Status.complete) {
+      if (state.status == Status.running) {
+        if (!_testCases.containsKey(liveTest.test)) {
+          String testName = liveTest.individualName;
+          testName = testName.replaceAll('&', '&amp;');
+          testName = testName.replaceAll('<', '&lt;');
+          testName = testName.replaceAll('>', '&gt;');
+          testName = testName.replaceAll('"', '&quot;');
+          testName = testName.replaceAll("'", '&apos;');
+
+          _testCases[liveTest] = new XunitTestResult(
+              testName, liveTest.suite.path, _stopwatch.elapsedMilliseconds);
+          List listOfGroups = _orderedGroupList(liveTest.groups);
+          Map listGroup = _groupStructure;
+
+          listOfGroups.forEach((elem) {
+            if (!listGroup.containsKey(elem)) {
+              listGroup[elem] = {'testResults': new XunitTestSuite()};
+            }
+            listGroup = listGroup[elem];
+          });
+
+          listGroup['testResults'] ??= new XunitTestSuite();
+
+          listGroup['testResults'].add(_testCases[liveTest]);
+        }
+      }
+      return;
+    }
+
+    _testCases[liveTest].endTime = _stopwatch.elapsedMilliseconds;
+
+    if (liveTest.test.metadata.skip) {
+      _testCases[liveTest].skipped = true;
+    }
+
+    if (liveTest.test.metadata.skipReason != null) {
```

Comment:
if this only applies when skipped is true, this can go inside the above if statement